Source/WebCore/bindings/js/JSDOMConvertStrings.cpp

```diff
@@ -118,4 +121,94 @@ AtomString valueToUSVAtomString(JSGlobalObject& lexicalGlobalObject, JSValue val
     return replaceUnpairedSurrogatesWithReplacementCharacter(WTFMove(string));
 }
 
+String valueToTrustedHTMLString(JSGlobalObject& lexicalGlobalObject, JSValue value, const String& sink, bool nullToEmpty)
+{
+    UNUSED_PARAM(sink);
+    VM& vm = lexicalGlobalObject.vm();
+
+    if (auto* trustedHTML = JSTrustedHTML::toWrapped(vm, value))
+        return trustedHTML->toString();
+
+    // TODO: Implement enforcement logic
+    if (value.isNull() && nullToEmpty)
+        return emptyString();
+
+    return nullString();
+}
+
+AtomString valueToTrustedHTMLAtomString(JSGlobalObject& lexicalGlobalObject, JSValue value, const String& sink, bool nullToEmpty)
+{
+    UNUSED_PARAM(sink);
+    VM& vm = lexicalGlobalObject.vm();
+
+    if (auto* trustedHTML = JSTrustedHTML::toWrapped(vm, value))
+        return trustedHTML->toString().toExistingAtomString();
```

Comment:
I don’t understand why the `AtomString` is guaranteed to exist here. Not sure that `toString().toExistingAtomString()` is the correct idiom for what we are trying to do.

Reply:
Ah apologies I forgot about this I meant to check what the correct way to convert from a string to an Atom String was.

Reply:
Given a String what is the correct way to convert it to an AtomString?

Reply:
`AtomString { string }` simply

Reply:
Thanks don't know how I missed that